### PR TITLE
Low-level Constructors

### PR DESCRIPTION
**Pull request by jmkuhn**

### CODE REVIEW

**Comment by jmkuhn** [comment]
Closes #14.

**Comment by jmkuhn** [comment]
All of the CI failures were errors installing Julia.  The runs where Julia installed, CI passed.

File: src/DecFP.jl [hunk 1]
```diff
@@ -566,6 +566,43 @@ function Base.decompose(x::DecimalFloatingPoint)::Tuple{BigInt, Int, BigInt}
     end
 end
 
+"""
+    Dec32(x::Union{Real, AbstractString} [, mode::RoundingMode])
+    Dec32([sign::Integer,] significand::Integer, exponent::Integer)
+"""
+function Dec32(sign::Integer, significand::Integer, exponent::Integer)
```
Comment (stevengj) [inline src/DecFP.jl:573]: Can we unify these functions?  e.g.
```jl
for (T,Ti) in ((Dec32,Int32), (Dec64,Int64))
    @eval function $(Symbol(T))(sign::Integer, significand::Integer, exponent::Integer)
        ...
    end
end
```
with the various constants interpolated as needed?

Comment (jmkuhn) [reply src/DecFP.jl:573]: Yes, I'll give this a try later today.

File: src/DecFP.jl [hunk 2]
```diff
@@ -124,6 +124,81 @@ for w in (32,64,128)
...
+        while s >= $Ti(maxintfloat($BID))
+            q, r = divrem(s, $Ti(10))
+            r != 0 && throw(InexactError(Symbol($BID), $BID, (sign, significand, exponent)))
+            s = q
+            e += 1
+        end
+        while e > bemax && s < $Ti(maxintfloat($BID)) / 10
+            s *= $Ti(10)
+            e -= 1
+        end
```
Comment (stevengj) [inline src/DecFP.jl:149]: Is this loop guaranteed to terminate?

Comment (stevengj) [reply src/DecFP.jl:149]: Oh, looks good.

**Comment by stevengj** [comment]
👍

**Pull request merged**